### PR TITLE
ISPN-13666 RemoteStore leaks out netty event loop client threads

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-13666

### CODE REVIEW

File: persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java [hunk 1]
```diff
@@ -283,7 +289,7 @@ public Publisher<MarshallableEntry<K, V>> publishEntries(IntSet segments, Predic
          if (filter != null) {
             entryFlowable = entryFlowable.filter(e -> filter.test(wrap(e.getKey())));
          }
-         return entryFlowable.map(e -> {
+         return entryFlowable.observeOn(Schedulers.from(nonBlockingExecutor)).map(e -> {
```
Comment (jabolina) [inline persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java:292]: Missing the return on the else branch.

Comment (wburns) [reply persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java:292]: :+1: will add in just a few mins here.

**Comment by wburns** [comment]
Updated

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Integrated, thanks @wburns!

**Pull request merged**